src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs

```diff
@@ -2538,6 +2538,161 @@ class C
             Assert.Equal(expectedText, text);
         }
 
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemove_KeepEndOfLine()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"
+class C
+{
+// before
+void M()
+{
+} // after
+}");
+
+            var expectedText = @"
+class C
+{
+
+}";
+
+            var m = cu.DescendantNodes().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+            Assert.NotNull(m);
+
+            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepEndOfLine);
+
+            var text = cu2.ToFullString();
+
+            Assert.Equal(expectedText, text);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveWithoutEOL_KeepEndOfLine()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"class A { } class B { } // test");
+
+            var m = cu.DescendantNodes().OfType<TypeDeclarationSyntax>().LastOrDefault();
+            Assert.NotNull(m);
+
+            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepEndOfLine);
+
+            var text = cu2.ToFullString();
+
+            Assert.Equal("class A { } ", text);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveBadDirectiveWithoutEOL_KeepEndOfLine_KeepDirectives()
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"class A { } class B { } #endregion");
+
+            var m = cu.DescendantNodes().OfType<TypeDeclarationSyntax>().LastOrDefault();
+            Assert.NotNull(m);
+
+            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepEndOfLine | SyntaxRemoveOptions.KeepDirectives);
+
+            var text = cu2.ToFullString();
+
+            Assert.Equal("class A { } #endregion", text);
```

Comment:
is this behavior changed from exiting one?

Reply:
Yes it is, the old code adds an EOL: `class A { } \r\n#endregion`
Similar as in [this case](https://github.com/dotnet/roslyn/pull/23165/files#diff-0e0e5cee589dd5835bc357a2c0406c46R2583).